tests/conftest.py

```diff
@@ -396,3 +397,98 @@ def log_cli_level(pytestconfig):
 
     """
     return pytestconfig.getini('log_cli_level') or 'DEBUG'
+
+
+@pytest.fixture(scope="class")
+def run_io_in_background(request):
+    """
+    Run IO during the test execution
+    """
+    if config.RUN['cli_params'].get('io_in_bg'):
+
+        lock = threading.Lock()
+        with lock:
+
+            # test_running is used for communication between the main thread
+            # and the new thread used for running the IO.
+            # Keeping test_running in a list because bool is immutable.
+            # Locking is used here as a synchronization mechanism to guard
+            # access to test_running.
+            request.test_running = [True]
+
+        results = list()
+
+        def finalizer():
+            """
+            Delete the resources created during setup, used for
+            running IO in the test background
+            """
+            with lock:
+                request.test_running = [False]
+            thread.join()
+            for result in results:
+                log.info(f"IOPs after FIO for pod {request.pod_obj.name}:")
+                log.info(
+                    f"Read: {result.get('jobs')[0].get('read').get('iops')}"
+                )
+                log.info(
+                    f"Write: {result.get('jobs')[0].get('write').get('iops')}"
+                )
+            if hasattr(request, 'pod_obj'):
+                request.pod_obj.delete()
+                request.pod_obj.ocp.wait_for_delete(
+                    resource_name=request.pod_obj.name
+                )
+            if hasattr(request, 'pvc_obj'):
+                request.pvc_obj.delete()
+                request.pvc_obj.ocp.wait_for_delete(
+                    resource_name=request.pvc_obj.name
+                )
+            if hasattr(request, 'sc_obj'):
+                request.sc_obj.delete()
+            if hasattr(request, 'secret_obj'):
+                request.secret_obj.delete()
+
+        request.addfinalizer(finalizer)
+
+        request.secret_obj = helpers.create_secret(
+            interface_type=constants.CEPHFILESYSTEM
+        )
+        request.sc_obj = helpers.create_storage_class(
+            interface_type=constants.CEPHFILESYSTEM,
+            interface_name=helpers.get_cephfs_data_pool_name(),
+            secret_name=request.secret_obj.name
+        )
+        request.pvc_obj = helpers.create_pvc(
+            sc_name=request.sc_obj.name, size='2Gi'
+        )
+        request.pod_obj = helpers.create_pod(
+            interface_type=constants.CEPHBLOCKPOOL,
+            pvc_name=request.pvc_obj.name
+        )
```

Comment:
The cli option has a side effect of creating extra objects for every test I believe, shouldn't this be just a test specific decorator?

Reply:
the option to run io in the BG is not related to any specific test. It is meant to provide a way to execute our test suites while the system is under some IO load

Reply:
yeah I got that, but it is applicable to specific set of tests and using the option for others will have side effect.

Reply:
I think that this is what we are looking for right?
To test everything in a lab environment is ok but not interesting, this adds more realistic aspects to the tests and with that we will be able to compare "lab" vs "real"